test/inject-test-mode

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+# This script injects a "test-mode" file into a copy of the main disk image. 
+# It begins by parsing the partition table of the disk image to identify the
+# 'aux' partition, which is then extracted. An empty 'test-mode' file is 
+# injected into the extracted partition, and the modified partition is written
+# back to the output image. The output image can subsequently be used as a 
+# backing image for Copy-on-Write (QCOW) images utilized by Qeneth.
+
+while getopts "b:o:" opt; do
+  case $opt in
+    b) base_img="$OPTARG" ;;  # Base image (Original image)
+    o) output_img="$OPTARG" ;;  # Output image (Backing image for QCoW images)
+    *) echo "Usage: $0 -b <BASE-IMAGE> -o <OUTPUT-IMAGE>" ; exit 1 ;;
+  esac
+done
+
+if [ -z "$base_img" ] || [ -z "$output_img" ]; then
+  echo "Both -b (base image) and -o (output image) parameters are required."
+  exit 1
+fi
+
+rm -f "$output_img"
+cp "$base_img" "$output_img"
+
+part_table=$(fdisk -l "$output_img")
```

Comment:
This is very optimistic. What if the copy fails, or if the file does not have a GPT partition table, or one without an `aux` partition? I think some error checking is warranted here. You might also want to `set -e` at the top.

Reply:
Yeah, good catch. Considered all of them.